app/src/main/java/kth/jjve/xfran/adapters/WorkoutsRecyclerAdapter.java

```diff
@@ -0,0 +1,165 @@
+    //plan workout
+    private final PlanButtonClickListener mPlanClickListener;
+    //save result for the workout
+    private final SaveButtonClickListener mSaveClickListener;
+
+    public interface ListItemClickListener{
+        //interface to expand workout item
+        void onListItemClick(int position);
+    }
+
+    public interface PlanButtonClickListener{
+        //interface to plan the workout
+        void onPlanButtonClick(int position);
+    }
+
+    public interface SaveButtonClickListener{
+        //interface to save the workout
+        void onSaveButtonClick(int position);
+    }
+
+    public WorkoutsRecyclerAdapter(Context context, List<Workout> workouts,
+                                   ListItemClickListener onItemClickListener,
+                                   PlanButtonClickListener onPlanButtonClickListener,
+                                   SaveButtonClickListener onSaveButtonClickListener) {
+        //adapter constructor receives the workout list and the listeners
+        mWorkouts = workouts;
+        mContext = context;
+        //set all workout items to collapsed view when adapter is constructed (ArrayList of false)
+        mExpandedStatus.clear();
+        mExpandedStatus.addAll(Collections.nCopies(mWorkouts.size(), false));
+        //initialize listeners
+        this.mItemClickListener = onItemClickListener;
+        this.mPlanClickListener = onPlanButtonClickListener;
+        this.mSaveClickListener = onSaveButtonClickListener;
+    }
+
+    @NonNull
+    @Override
+    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
+        View view = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.workout_item, viewGroup, false);
+        return new ViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder vh, int i) {
+
+        //Set the name of the 'Workout'
+        ((ViewHolder) vh).mName.setText(mWorkouts.get(i).getTitle());
+        //Set the details of the 'Workout'
+        ((ViewHolder) vh).mDescription.setText(mWorkouts.get(i).getType());
+        //Set the exercises of the 'Workout'
+        String exercises = "";
+        ArrayList<String> exercisesArray = mWorkouts.get(i).getExercises();
+        for (String s : exercisesArray){
+            exercises += s + "\n";
+        }
+        ((ViewHolder) vh).mExercises.setText(exercises);
+
+        //Set workout item view to collapsed/expanded according to ArrayList value
+        boolean expanded = mExpandedStatus.get(i);
+        ((ViewHolder) vh).mExpandedView.setVisibility(expanded ? View.VISIBLE : View.GONE);
+        mExpandedStatus.set(i, (!expanded));
+    }
+
+    @Override
+    public int getItemCount() {
+        return mWorkouts.size();
+    }
+
+    private class ViewHolder extends RecyclerView.ViewHolder {
+
+        /*------ HOOKS ------*/
+        private TextView mName;
+        private TextView mDescription;
```

Comment:
combine all the textviews, buttons and views :) Secondly it is always a good practice to keep an order (so not:

1. textviews
2. view
3. textviews
4. button
But rather
1. textviews
2. views
3. buttons